ignite/services/plugin/template/main.go.plush

```diff
@@ -3,118 +3,47 @@ package main
 import (
 	"context"
 	"fmt"
-	"path/filepath"
 
 	hplugin "github.com/hashicorp/go-plugin"
 
-	"github.com/ignite/cli/v28/ignite/services/chain"
 	"github.com/ignite/cli/v28/ignite/services/plugin"
+	"<%= ModuleName %>/cmd"
 )
 
 type app struct{}
 
-func (app) Manifest(ctx context.Context) (*plugin.Manifest, error) {
+func (app) Manifest(_ context.Context) (*plugin.Manifest, error) {
 	return &plugin.Manifest{
-		Name: "<%= Name %>",
-		// TODO: Add commands here
-		Commands: []*plugin.Command{
-			// Example of a command
-			{
-				Use:   "<%= Name %>",
-				Short: "Explain what the command is doing...",
-				Long:  "Long description goes here...",
-				Flags: []*plugin.Flag{
-					{Name: "my-flag", Type: plugin.FlagTypeString, Usage: "my flag description"},
-				},
-				PlaceCommandUnder: "ignite",
-				// Examples of adding subcommands:
-				// Commands: []*plugin.Command{
-				// 	{Use: "add"},
-				// 	{Use: "list"},
-				// 	{Use: "delete"},
-				// },
-			},
-		},
-		// TODO: Add hooks here
-		Hooks: []*plugin.Hook{},
-		SharedHost: <%= SharedHost %>,
+		Name: "<%= Name %>",<%= if (SharedHost) { %>
+		SharedHost: true,<% } %>
+		Commands: cmd.GetCommands(),
 	}, nil
 }
 
-func (app) Execute(ctx context.Context, cmd *plugin.ExecutedCommand, api plugin.ClientAPI) error {
-	// TODO: write command execution here
-	fmt.Printf("Hello I'm the example-plugin plugin\n")
-	fmt.Printf("My executed command: %q\n", cmd.Path)
-	fmt.Printf("My args: %v\n", cmd.Args)
-
-	flags, err := cmd.NewFlags()
-	if err != nil {
-		return err
-	}
-
-	myFlag, _ := flags.GetString("my-flag")
-	fmt.Printf("My flags: my-flag=%q\n", myFlag)
-	fmt.Printf("My config parameters: %v\n", cmd.With)
-
-	// This is how the plugin can access the chain:
-	// c, err := getChain(cmd)
-	// if err != nil {
-	// 	return err
-	// }
-
-	// According to the number of declared commands, you may need a switch:
-	/*
-		switch cmd.Use {
-		case "add":
-			fmt.Println("Adding stuff...")
-		case "list":
-			fmt.Println("Listing stuff...")
-		case "delete":
-			fmt.Println("Deleting stuff...")
-		}
-	*/
+func (app) Execute(ctx context.Context, c *plugin.ExecutedCommand, _ plugin.ClientAPI) error {
+	// Remove the first two elements "ignite" and "<%= Name %>" from OsArgs.
+	args := c.OsArgs[2:]
 	
-	// ClientAPI call example
-	fmt.Println(api.GetChainInfo(ctx))
-	
-	return nil
+	switch args[0] {
+	case "hello":
+		return cmd.ExecuteHello(ctx, c)
+	default:
+		return fmt.Errorf("unknown command: %s", c.Path)
```

Comment:
```suggestion
		return fmt.Errorf("unknown command '%s'; valid commands are [hello]", args[0])
```

Reply:
Let's keep it simple for now, because here we are only trying to scaffold a base template of an app for developers. In Future we can develop helper functions that do these stuff without boilerplates.